resources/osl_ssh_key.rb

```diff
@@ -0,0 +1,40 @@
+provides :osl_ssh_key
+resource_name :osl_ssh_key
+unified_mode true
+
+default_action :add
+
+property :key_name, String, name_property: true
+property :key, String, required: [:add]
+property :user, String, required: true
+property :group, String, default: lazy { user }
```

Comment:
Does using new_resource work here?

```suggestion
property :group, String, default: lazy { new_resource.user }
```

Reply:
probably, is there a reason for that?

Reply:
`new_resource` is explicitly scoped to the resource properties and makes it easier to tell thats where the value is coming from, but a while ago I remember `new_resource` was not available outside of the actions. If it works here, that would be preferred over the bare property.

Reply:
doesnt work actually